Add stable releases of ctlr, ingress

recursivelycurious opened the pull request:
Problem:
- Update charts to include necessary documentation
- Provide stable ctlr, ingress charts
- Update examples based on regression testing

Solution:
- Updated and expanded documentation in README.md files
- Updated and expanded documentation in values files
- Expanded ctlr chart ability to accept additional parameters
- Modified argument parameter naming convetntion to offer more consistency with product docs
- Fixed typos in example files
- Packaged tars, reindexed incubator and stable

Testing:
- Manually tested modified charts in k8s cluster
- Nightly regression testing for all examples is in place

fixes #1

recursivelycurious commented:
@jputrino -- some info to simplify review...

To see the documentation essentially as the user would:
- https://github.com/recursivelycurious/charts/tree/stable-ingress
- https://github.com/recursivelycurious/charts/tree/stable-ingress/src/stable/f5-bigip-ctlr
- https://github.com/recursivelycurious/charts/tree/stable-ingress/src/stable/f5-bigip-ingress

With the caveat that any links back into the Chart repo itself will not work until the merge is complete.

In regards to the index and tar files -- A user of helm gets local versions of the charts via the archived files. We offer two "repos", stable and incubator, and each folder has its own index which is what permits github pages to serve the  content of those folders in a way that helm can consume.

Note that we maintain distinct `src` files for incubator and stable for a number of reasons, but one good example is that the incubator points to the devel-master image for the controller and the stable points to production.

Feel free to review the docs portion only if cycles are low and I can have another dev review the rest if needed.

Thanks!

recursivelycurious commented:
@jputrino - changed to a wip to confirm that tars for forced commit include changes. Review for documentation would still be helpful while I confirm/resolve that issue. Thanks!

jputrino commented on README.md:
````diff
@@ -1,27 +1,30 @@
 ```
 
-To access additional charts in beta:
+These charts are tested and documented:
````
Stolen from the openstack heat project repos, feel free to amend or disregard as appropriate:
```
The stable directory contains charts that are created and tested by F5 Networks. These templates are supported by F5 (see [SUPPORT](./SUPPORT) for details).
```
Also, add the SUPPORT file if appropriate (maybe check with Ashok?) - https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/SUPPORT.md

recursivelycurious replied:
Used part of the language and will ask Ashok about support.

recursivelycurious replied:
Adding the support reference per offline conversation with @amudukutore

jputrino commented on README.md:
```diff
@@ -1,27 +1,30 @@
-To access additional charts in beta:
+These charts are tested and documented:
+- [f5-bigip-ctlr](https://github.com/F5Networks/charts/tree/master/src/stable/f5-bigip-ctlr) - Use this chart to deploy the [k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/) in Kubernetes or OpenShift.
+- [f5-bigip-ingress](https://github.com/F5Networks/charts/tree/master/src/stable/f5-bigip-ingress) - Use this chart to deploy ingress resources in Kubernetes or OpenShift.
```
Capitalize "Ingress"

recursivelycurious replied:
done

jputrino commented on README.md:
```diff
@@ -1,27 +1,30 @@
+## Documentation
 
-## Repository Structure
+Each chart contains its own `README.md` and the `values.yaml` file includes notes on default values and links to the core documentation for the resources each chart is deploying.
```
Each chart has a README describing its basic functionality. The `values.yaml` file for each chart shows the default values and contains links to documentation for the resources the chart deploys.

recursivelycurious replied:
done

jputrino commented on README.md:
```diff
@@ -1,27 +1,30 @@
+## Incubation Charts
 
-Charts in `/stable` have been tested and documented, charts in `/incubator` may not yet be fully tested and documented. 
+To access additional charts in a development or testing mode that may not be documented:
```
Again, taken from the f5 heat repos, feel free to amend/ignore:
```
The incubation charts may have been created by F5 Networks or by external contributors. These charts have not undergone full testing and are subject to change. F5 Networks does not provide technical support for templates in the incubation directory.

To access charts in the incubation area: 

```

recursivelycurious replied:
Like it. Used it. Done :)

jputrino commented on src/incubator/f5-bigip-ctlr/Chart.yaml:
```diff
@@ -1,4 +1,4 @@
 apiVersion: v1
-description: Deploy the F5 Networks k8s-bigip-ctlr 
+description: Deploy the F5 Networks k8s-bigip-ctlr for using the BIG-IP in a Kubernetes or OpenShift environment
```
Deploy the F5 Networks BIG-IP Controller for Kubernetes and OpenShift (k8s-bigip-ctlr)

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -1,11 +1,25 @@
-- The BigIP management url must be passed as an argument using `--set` or configured in a values file.
+## Requirements
+
+The chart presumes:
```
I would change this to a Prerequisites sub-section

```
### Prerequisites
- Add your BIG-IP device to your [Kubernetes](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-use-bigip-k8s.html) or [OpenShift](http://clouddocs.f5.com/containers/v2/openshift/kctlr-use-bigip-openshift.html) Cluster.
- Create a partition on your BIG-IP device for the BIG-IP Controller to manage. The Controller cannot manage objects in the `/Common` partition.
- Create a Secret containing the BIG-IP login credentials for the Controller. The Controller needs an account with administrator-level permissions to ensure full functionality.

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -1,11 +1,25 @@
+- A partition (other than common) exists on your BIG-IP for the instance of the controller to manage.
+- A secret exists in your cluster to access the BIG-IP.
+
+The default values for the partition and the secret to access the BIG-IP are:
```
The chart contains the following default values for partition and Secret, respectively.

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -1,11 +1,25 @@
+- `f5-bigip-ctlr` and 
+- `f5-bigip-ctlr-login` 
+
+These values can be easily updated using either `--set <param>=<value>` or `-f <values-file.yaml>`. See [customizing the chart before installing](https://docs.helm.sh/using_helm/#customizing-the-chart-before-installing) for more details.
```
Be sure to change these if they differ from your actual partition and Secret names, using `--set <param>=<value>` or `-f <values-file.yaml>` as appropriate.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -1,11 +1,25 @@
+
+## Chart Details
+
+The chart creates a Deployment for one Pod containing the [k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/) and the supporting RBAC resources.
```
'the supporting' --> 'its supporting'

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -1,11 +1,25 @@
+
+## Installing the Chart
 
 If using the names in the default values;
```
Run the command shown below to install the chart using the default values.

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/templates/NOTES.txt:
```diff
@@ -1 +1,11 @@
+- Kubernetes: http://clouddocs.f5.com/containers/latest/kubernetes/index.html
+- OpenShift: http://clouddocs.f5.com/containers/latest/openshift/index.html
+
+Using Ingress? There is a helm chart for that:
```
There is --> there's

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/templates/NOTES.txt:
```diff
@@ -1 +1,11 @@
+Using Ingress? There is a helm chart for that:
+- https://github.com/F5Networks/charts/tree/master/src/incubator/f5-bigip-ingress
+
+Using Routes in OpenShift? No helm chart yet, but we do have great documentation:
```
love it :)

recursivelycurious replied:
:)

jputrino commented on src/incubator/f5-bigip-ctlr/values.yaml:
```diff
@@ -1,21 +1,46 @@
+# http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-secrets.html#secret-bigip-login
+bigip-login-secret: f5-bigip-ctlr-login
 serviceaccount: f5-bigip-ctlr-serviceaccount
+# This namespace is where the controller lives;
```
Controller

recursivelycurious replied:
Done

jputrino commented on src/incubator/f5-bigip-ctlr/values.yaml:
```diff
@@ -1,21 +1,46 @@
+  # REQUIRED Params
+  bigip-url: ~
+  bigip-partition: f5-bigip-ctlr
+  # OPTIONAL PARAMS -- simply uncomment and provide values for those you wish to use.
```
remove "simply"

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -1,12 +1,25 @@
-# alpha status
+# Helm Chart for using the F5 BIG-IP as an ingress
```
Suggested edit: using the F5 BIG-IP as an ingress --> Managing Ingress Resources with a BIG-IP Device

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -1,12 +1,25 @@
+# Helm Chart for using the F5 BIG-IP as an ingress
 
-This chart is in an alpha status. It can be used to deploy an ingress with on simple service. 
+This chart simplifies repeatable, versioned use of the F5 BIG-IP as an [ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) within Kubernetes or OpenShift by using the [k8s-bigip-ctlr](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-ingress.html) as an ingress controller.
```
"as an ingress" --> "to manage Ingress resources ..."

or, change the whole sentence to
```
"This chart simplifies repeatable, versioned use of the [F5 BIG-IP Controller as an Ingress Controller](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-k8s-ingress-ctlr.html) in Kubernetes or OpenShift. "
```
(link points to new doc coming out with grays release)

recursivelycurious replied:
Used suggested alternative, thanks. :)

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -1,12 +1,25 @@
+## Requirements
 
-# Suggested Use
+The chart presumes:
```
Same as above - reformat as Prerequisites section

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -1,12 +1,25 @@
+
+## Installing the Chart
+
+Copy the `values.yaml` file and use it as the basis for your own ingress specification and then pass your custom values file when invoking `helm install`:
```
1. Copy the values file. Use this file as the basis for your own Ingress specification.
2. Pass your custom values file when running helm install, as shown in the example below.

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +33,20 @@ Or
+
+### Additional Optional parameters as annotations
+
+The annotations listed under the `ingress.annotations` parameter are consumed as an array and any of the [documentend annotations for the k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#supported-ingress-annotations) may be used.
```
The Controller consumes the `ingress.annotations` as a JSON array. (correct?) You can provide any of the k8s-bigip-ctlr supported Ingress Annotations.

recursivelycurious replied:
The health monitors themselves have to be provided as a JSON array - YAML will not work (though it should https://github.com/F5Networks/k8s-bigip-ctlr/issues/594). I added a bit more clarity to this section.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +33,20 @@ Or
+
+The annotations listed under the `ingress.annotations` parameter are consumed as an array and any of the [documentend annotations for the k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#supported-ingress-annotations) may be used.
+
+> CAUTION: Be sure to use a controller version compatable with the annotations you choose.
```
Be sure to use the correct version of the Controller for the Annotations you wish to use. See the [k8s-bigip-ctlr release notes](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/RELEASE-NOTES.html) for more information.

recursivelycurious replied:
Done

jputrino commented on src/incubator/f5-bigip-ingress/values.yaml:
```diff
@@ -1,5 +1,9 @@
 # Values template for f5-bigip-ingress
 # Suggested use is to copy, specify values, and pass with `-f` to helm install
+# Additional information on using the k8s-bigip-ctlr as an ingress controller
```
Ingress

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/Chart.yaml:
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+description: Deploy the F5 Networks k8s-bigip-ctlr for using the BIG-IP in a Kubernetes or OpenShift environment
```
same as above - Deploy the F5 Networks BIG-IP Controller for Kubernetes and OpenShift (k8s-bigip-ctlr)

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,53 @@
+
+## Requirements
+
+The chart presumes:
```
Same as above - format as Prerequisites

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,53 @@
+
+## Chart Details
+
+The chart creates a Deployment for one Pod containing the [k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/) and the supporting RBAC resources.
```
the supporting --> its supporting

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -1,12 +1,25 @@
+The chart presumes:
+- You have [Helm with Tiller](https://docs.helm.sh/using_helm/#installing-helm) installed on your cluster with appropriate permissions.
+- You have the k8s-bigip-ctlr already running in your cluster. You can use a separate chart - the [f5-bigip-ctlr chart](https://github.com/F5Networks/charts/tree/master/src/stable/f5-bigip-ctlr) - to deploy the controller, or you can deploy it [manually](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-app-install.html). 
+- The services accepting traffic from the ingress are configured separately.
```
Ingress

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,53 @@
+
+## Installing the Chart
+
+If using the names in the default values;
```
Run the command shown below to install the chart using the default values.

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,53 @@
+args.verify-interval | Interval in seconds to verify BIG-IP | 2 for incubation
+args.node-poll-interval | Interval in seconds to poll the cluster | 1 for incubation
+
+See the Controller documentation for a full list of [configuration parameters](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/v1.4/#controller-configuration-parameters).
```
v1.4 --> latest

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/templates/NOTES.txt:
```diff
@@ -0,0 +1,11 @@
+- Kubernetes: http://clouddocs.f5.com/containers/latest/kubernetes/index.html
+- OpenShift: http://clouddocs.f5.com/containers/latest/openshift/index.html
+
+Using Ingress? There is a helm chart for that:
```
There is --> There's

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/templates/f5-bigip-ctlr-clusterrole.yaml:
```diff
@@ -0,0 +1,39 @@
+  - endpoints
+  - namespaces
+  - ingresses
+  - secrets
```
Up for debate - consider naming a specific secret instead of granting access to all Kubernetes secrets cluster-wide. I've made this update in the solution doc examples (see https://github.com/F5Networks/f5-ci-docs/pull/401/commits/70798ca108212dcf45c170a3ec8a6a999fa1edcd)

recursivelycurious replied:
I'll need to do some testing to ensure that this change, so not including in this revision.

jputrino commented on src/stable/f5-bigip-ingress/README.md:
```diff
@@ -0,0 +1,52 @@
+# Helm Chart for using the F5 BIG-IP as an ingress
```
Ingress

Same as previous comments re using BIG-IP device to manage Ingress resources

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ingress/README.md:
```diff
@@ -0,0 +1,52 @@
+
+## Requirements
+
+The chart presumes:
```
Format as Prerequisites list.

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ingress/README.md:
```diff
@@ -0,0 +1,52 @@
+
+## Installing the Chart
+
+Copy the `values.yaml` file and use it as the basis for your own ingress specification and then pass your custom values file when invoking `helm install`:
```
Ingress

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ingress/README.md:
```diff
@@ -0,0 +1,52 @@
+
+### Additional Optional parameters as annotations
+
+The annotations listed under the `ingress.annotations` parameter are consumed as an array and any of the [documentend annotations for the k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#supported-ingress-annotations) may be used.
```
Same updates as above

recursivelycurious replied:
Done.

jputrino replied:
annotations --> Annotations

recursivelycurious commented on README.md:
```diff
@@ -1,27 +1,30 @@
+This repository contains [helm](https://docs.helm.sh/using_helm/#using) charts for use with some [F5 Networks](https://f5.com/) products and services within a [Kubernetes](https://kubernetes.io/) or [OpenShift](https://www.openshift.com/) environment.
 
-# F5 Networks Helm Charts
+**Note** that charts may require access to `kube-system` namespace and/or cluster wide permissions to function as expected, so be sure to install/configure helm/tiller appropriately.
```
done

recursivelycurious commented:
@jputrino - All comments were addressed. In order to ensure that the tar files were current, had to re-roll them and force push. 

Below is the diff. Note that I copied incubator to stable, then made the handful of changes in stable that referenced dev or incubator resources or endpoints.

```
diff --git a/src/incubator/f5-bigip-ctlr/Chart.yaml b/src/incubator/f5-bigip-ctlr/Chart.yaml
index 581f7e7..e2bdaa9 100644
--- a/src/incubator/f5-bigip-ctlr/Chart.yaml
+++ b/src/incubator/f5-bigip-ctlr/Chart.yaml
@@ -1,4 +1,4 @@
 apiVersion: v1
-description: Deploy the F5 Networks k8s-bigip-ctlr for using the BIG-IP in a Kubernetes or OpenShift environment
+description: Deploy the F5 Networks BIG-IP Controller for Kubernetes and OpenShift (k8s-bigip-ctlr). 
 name: f5-bigip-ctlr
-version: 0.0.3
+version: 0.0.4
diff --git a/src/incubator/f5-bigip-ctlr/README.md b/src/incubator/f5-bigip-ctlr/README.md
index a0fbeec..b0708ef 100644
--- a/src/incubator/f5-bigip-ctlr/README.md
+++ b/src/incubator/f5-bigip-ctlr/README.md
@@ -2,26 +2,24 @@
 
 This chart simplifies repeatable, versioned deployment of the [F5 BIG-IP Controller for Kubernetes](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/).
 
-## Requirements
+### Prerequisites
+- Add your BIG-IP device to your [Kubernetes](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-use-bigip-k8s.html) or [OpenShift](http://clouddocs.f5.com/containers/v2/openshift/kctlr-use-bigip-openshift.html) Cluster.
+- Create a partition on your BIG-IP device for the BIG-IP Controller to manage. The Controller cannot manage objects in the `/Common` partition.
+- Create a Secret containing the BIG-IP login credentials for the Controller. The Controller needs an account with administrator-level permissions to ensure full functionality.
 
-The chart presumes:
-- You have added a BIG-IP device to either your [Kubernetes](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-use-bigip-k8s.html) or [OpenShift](http://clouddocs.f5.com/containers/v2/openshift/kctlr-use-bigip-openshift.html) cluster.
-- A partition (other than common) exists on your BIG-IP for the instance of the controller to manage.
-- A secret exists in your cluster to access the BIG-IP.
-
-The default values for the partition and the secret to access the BIG-IP are:
+The chart contains the following default values for partition and Secret, respectively:
 - `f5-bigip-ctlr` and 
 - `f5-bigip-ctlr-login` 
 
-These values can be easily updated using either `--set <param>=<value>` or `-f <values-file.yaml>`. See [customizing the chart before installing](https://docs.helm.sh/using_helm/#customizing-the-chart-before-installing) for more details.
+Be sure to change these if they differ from your actual partition and Secret names, using `--set <param>=<value>` or `-f <values-file.yaml>` as appropriate. See [customizing the chart before installing](https://docs.helm.sh/using_helm/#customizing-the-chart-before-installing) for more details.
 
 ## Chart Details
 
-The chart creates a Deployment for one Pod containing the [k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/) and the supporting RBAC resources.
+The chart creates a Deployment for one Pod containing the [k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/) and its supporting RBAC resources.
 
 ## Installing the Chart
 
-If using the names in the default values;
+Run the commands shown below to install the chart using the default values.
 
 ```
 helm repo add f5-incubator https://f5networks.github.io/charts/incubator
@@ -37,7 +35,7 @@ helm install --set args.bigip_url=1.2.3.4 charts/src/incubator/f5-bigip-ctlr
 
 ## Chart parameters:
 
-> **note** Helm value names cannot include the character `-`, so parameter names in `values.yaml` use `_` and then replace them with `-` when rendering.
+> **Note:** Helm value names cannot include the character `-` which is commonly used in the names of parameters passed to the controller. To accomodate Helm, the parameter names in `values.yaml` use `_` and then replace them with `-` when rendering.
 > e.g. `args.bigip_url` is rendered as `bigip-url` as required by the Controler.
 
 
@@ -51,7 +49,7 @@ args.log_level | Log detail | DEBUG for incubation chart
 args.verify_interval | Interval in seconds to verify BIG-IP | 2 for incubation
 args.node_poll_interval | Interval in seconds to poll the cluster | 1 for incubation
 
-See the Controller documentation for a full list of [configuration parameters](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/v1.4/#controller-configuration-parameters).
+See the Controller documentation for a full list of [configuration parameters](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#controller-configuration-parameters).
 
 If you have a specific use case for F5 products in the kubernetes environment that would benefit from a curated chart, please [open an issue](https://github.com/F5Networks/charts/issues) describing your use case and providing example resources.
 
diff --git a/src/incubator/f5-bigip-ctlr/templates/NOTES.txt b/src/incubator/f5-bigip-ctlr/templates/NOTES.txt
index 5623419..ef77f73 100644
--- a/src/incubator/f5-bigip-ctlr/templates/NOTES.txt
+++ b/src/incubator/f5-bigip-ctlr/templates/NOTES.txt
@@ -4,7 +4,7 @@ General Controller Documentation:
 - Kubernetes: http://clouddocs.f5.com/containers/latest/kubernetes/index.html
 - OpenShift: http://clouddocs.f5.com/containers/latest/openshift/index.html
 
-Using Ingress? There is a helm chart for that:
+Using Ingress? There's a helm chart for that:
 - https://github.com/F5Networks/charts/tree/master/src/incubator/f5-bigip-ingress
 
 Using Routes in OpenShift? No helm chart yet, but we do have great documentation:
diff --git a/src/incubator/f5-bigip-ctlr/templates/f5-bigip-ctlr-deploy.yaml b/src/incubator/f5-bigip-ctlr/templates/f5-bigip-ctlr-deploy.yaml
index 8db87d4..36649bb 100644
--- a/src/incubator/f5-bigip-ctlr/templates/f5-bigip-ctlr-deploy.yaml
+++ b/src/incubator/f5-bigip-ctlr/templates/f5-bigip-ctlr-deploy.yaml
@@ -3,7 +3,7 @@
 Generate errors for missing required values.
 */}}
 # {{required "BIG-IP url not specified - add to Values or pass with `--set` " .Values.args.bigip_url }}
-# {{required "BIG-IP partition not specified - add to Values or pass with `--set` " .Values.args.partition }}
+# {{required "BIG-IP partition not specified - add to Values or pass with `--set` " .Values.args.bigip_partition }}
 {{- else -}}
 apiVersion: extensions/v1beta1
 kind: Deployment
diff --git a/src/incubator/f5-bigip-ctlr/values.yaml b/src/incubator/f5-bigip-ctlr/values.yaml
index dcaf677..8f94757 100644
--- a/src/incubator/f5-bigip-ctlr/values.yaml
+++ b/src/incubator/f5-bigip-ctlr/values.yaml
@@ -7,7 +7,7 @@
 # http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-secrets.html#secret-bigip-login
 bigip_login_secret: f5-bigip-ctlr-login
 serviceaccount: f5-bigip-ctlr-serviceaccount
-# This namespace is where the controller lives;
+# This namespace is where the Controller lives;
 namespace: kube-system
 args:
   # See http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#controller-configuration-parameters
@@ -16,9 +16,9 @@ args:
   # REQUIRED Params
   bigip_url: ~
   bigip_partition: f5-bigip-ctlr
+  # OPTIONAL PARAMS -- uncomment and provide values for those you wish to use.
   verify_interval: 2
   node-poll_interval: 1
-  # OPTIONAL PARAMS
   log_level: DEBUG
   # python_basedir: ~
   # VXLAN
diff --git a/src/incubator/f5-bigip-ingress/README.md b/src/incubator/f5-bigip-ingress/README.md
index ef79420..607c3e2 100644
--- a/src/incubator/f5-bigip-ingress/README.md
+++ b/src/incubator/f5-bigip-ingress/README.md
@@ -1,17 +1,17 @@
-# Helm Chart for using the F5 BIG-IP as an ingress
+# Helm Chart for Managing Ingress Resources with a BIG-IP Device
 
-This chart simplifies repeatable, versioned use of the F5 BIG-IP as an [ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) within Kubernetes or OpenShift by using the [k8s-bigip-ctlr](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-ingress.html) as an ingress controller.
+This chart simplifies repeatable, versioned use of the [F5 BIG-IP Controller as an Ingress Controller](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-k8s-ingress-ctlr.html) in Kubernetes or OpenShift. 
 
-## Requirements
+### Prereqisites
 
-The chart presumes:
-- You have [Helm with Tiller](https://docs.helm.sh/using_helm/#installing-helm) installed on your cluster with appropriate permissions.
-- You have the k8s-bigip-ctlr already running in your cluster. You can use a separate chart - the [f5-bigip-ctlr chart](https://github.com/F5Networks/charts/tree/master/src/stable/f5-bigip-ctlr) - to deploy the controller, or you can deploy it [manually](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-app-install.html). 
-- The services accepting traffic from the ingress are configured separately.
+- Install [Helm with Tiller](https://docs.helm.sh/using_helm/#installing-helm) on your cluster with appropriate permissions.
+- Deploy the F5 BIG-IP Controller in your cluster. You can use a separate chart - the [f5-bigip-ctlr chart](https://github.com/F5Networks/charts/tree/master/src/stable/f5-bigip-ctlr) - to deploy the controller, or you can deploy it [manually](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-app-install.html). 
+- Deploy the Pods/Services accepting traffic from the Ingress.
 
-> Note - This chart and the [f5-bigip-controller](https://github.com/recursivelycurious/charts/tree/wip/src/incubator/f5-bigip-ctlr) chart can be used independently or together. If you or your organization author your own charts either or both may be used as a subchart.
+> **Note:** This chart and the [f5-bigip-controller](https://github.com/recursivelycurious/charts/tree/wip/src/incubator/f5-bigip-ctlr) chart can be used *independently or together*.  
+> If you or your organization author your own charts either or both may be used as a [subchart](https://docs.helm.sh/chart_template_guide/#creating-a-subchart).
 >
-> Similarly, this ingress chart can be combined -- either as a parent chart or a subchart -- with charts that define the services accepting traffic.
+> Similarly, this Ingress chart can be combined -- either as a parent chart or a subchart -- with charts that define the services accepting traffic.
 
 ## Chart Details
 
@@ -19,7 +19,8 @@ The chart creates an Ingress resource for use with the [k8s-bigip-ctlr](http://c
 
 ## Installing the Chart
 
-Copy the `values.yaml` file and use it as the basis for your own ingress specification and then pass your custom values file when invoking `helm install`:
+1. Copy the `values.yaml` file. Use this file as the basis for your own Ingress specification. 
+2. Pass your custom values file when running `helm install` as shown in the example below.
 
 ```
 helm repo add f5-incubator https://f5networks.github.io/charts/incubator
@@ -46,7 +47,9 @@ spec | Backend(s) and associated hosts and paths | See [examples](https://github
 
 The annotations listed under the `ingress.annotations` parameter are consumed as an array and any of the [documentend annotations for the k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#supported-ingress-annotations) may be used.
 
-> CAUTION: Be sure to use a controller version compatable with the annotations you choose.
+When using the `virtual-server.f5.com/health` annotation the value must be a JSON array of the individual health monitors. Single and multiple health monitor examples can be seen in the [Ingress Examples](https://github.com/F5Networks/charts/tree/master/example_values) in this repo.
+
+> CAUTION: Be sure to use the correct version of the Controller for the annotations you wish to use. See the [k8s-bigip-ctlr release notes](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/RELEASE-NOTES.html) for more information.
 
 If you have a specific use case for F5 products in the kubernetes environment that would benefit from a curated chart, please [open an issue](https://github.com/F5Networks/charts/issues) describing your use case and providing example resources.
 
diff --git a/src/incubator/f5-bigip-ingress/values.yaml b/src/incubator/f5-bigip-ingress/values.yaml
index bb43c77..2741a2e 100644
--- a/src/incubator/f5-bigip-ingress/values.yaml
+++ b/src/incubator/f5-bigip-ingress/values.yaml
@@ -1,6 +1,6 @@
 # Values template for f5-bigip-ingress
 # Suggested use is to copy, specify values, and pass with `-f` to helm install
-# Additional information on using the k8s-bigip-ctlr as an ingress controller
+# Additional information on using the k8s-bigip-ctlr as an Ingress controller
 # within Kubernetes/Openshift can be found at
 # http://clouddocs.f5.com/containers/v2/kubernetes/kctlr-ingress.html
 #

```

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -21,4 +33,23 @@ Or
+----------|-------------|--------
+bigip_login_secret | Secret that contains BIG-IP login credentials | f5-bigip-ctlr-login
+serviceaccount | name of ServiceAccount the ctlr should use | f5-bigip-ctlr-serviceaccount
+args.bigip_url | The admin IP for your BIG-IP | **Required**, no default
```
"admin IP" --> "management IP"

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -21,4 +33,23 @@ Or
+bigip_login_secret | Secret that contains BIG-IP login credentials | f5-bigip-ctlr-login
+serviceaccount | name of ServiceAccount the ctlr should use | f5-bigip-ctlr-serviceaccount
+args.bigip_url | The admin IP for your BIG-IP | **Required**, no default
+args.partition | BIG-IP partition the ctlr is to manage | f5-bigip-ctlr
```
"is to" --> "should" (or "will")

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -21,4 +33,23 @@ Or
+args.bigip_url | The admin IP for your BIG-IP | **Required**, no default
+args.partition | BIG-IP partition the ctlr is to manage | f5-bigip-ctlr
+args.log_level | Log detail | DEBUG for incubation chart
+args.verify_interval | Interval in seconds to verify BIG-IP | 2 for incubation
```
Interval, in seconds, at which to verify BIG-IP settings

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -21,4 +33,23 @@ Or
+args.partition | BIG-IP partition the ctlr is to manage | f5-bigip-ctlr
+args.log_level | Log detail | DEBUG for incubation chart
+args.verify_interval | Interval in seconds to verify BIG-IP | 2 for incubation
+args.node_poll_interval | Interval in seconds to poll the cluster | 1 for incubation
```
Interval, in seconds, at which to poll the cluster

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/values.yaml:
```diff
@@ -1,21 +1,48 @@
 namespace: kube-system
+args:
+  # See http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#controller-configuration-parameters
+  # NOTE: helm has difficulty with values using `-`; `_` are used for nameing
```
"nameing" --> "naming"

recursivelycurious replied:
Done

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -1,12 +1,26 @@
 
-# Suggested Use
+- Install [Helm with Tiller](https://docs.helm.sh/using_helm/#installing-helm) on your cluster with appropriate permissions.
+- Deploy the F5 BIG-IP Controller in your cluster. You can use a separate chart - the [f5-bigip-ctlr chart](https://github.com/F5Networks/charts/tree/master/src/stable/f5-bigip-ctlr) - to deploy the controller, or you can deploy it [manually](http://clouddocs.f5.com/containers/latest/kubernetes/kctlr-app-install.html). 
```
You can use the [f5-bigip-ctlr](...) chart to deploy the Controller or deploy it [manually](...).

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -1,12 +1,26 @@
+> **Note:** This chart and the [f5-bigip-controller](https://github.com/recursivelycurious/charts/tree/wip/src/incubator/f5-bigip-ctlr) chart can be used *independently or together*.  
+> If you or your organization author your own charts either or both may be used as a [subchart](https://docs.helm.sh/chart_template_guide/#creating-a-subchart).
+>
+> Similarly, this Ingress chart can be combined -- either as a parent chart or a subchart -- with charts that define the services accepting traffic.
```
"services" --> "Services"

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +34,22 @@ Or
+
+Parameter | Description | Default
+----------|-------------|--------
+ingress.annotations.virtual-server.f5.com/ip | IP accepting traffic on the BIG-IP | **Required** no default
```
"the BIG-IP system" or "the BIG-IP device"

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +34,22 @@ Or
+----------|-------------|--------
+ingress.annotations.virtual-server.f5.com/ip | IP accepting traffic on the BIG-IP | **Required** no default
+ingress.annotations.virtual-server.f5.com/partition | BIG-IP partition of the Controller | **Required** no default
+ingress.namespace | Kubernetes/OpenShift namespace for the ingress | Optional
```
"ingress" --> "Ingress"

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +34,22 @@ Or
+
+### Additional Optional parameters as annotations
+
+The annotations listed under the `ingress.annotations` parameter are consumed as an array and any of the [documentend annotations for the k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#supported-ingress-annotations) may be used.
```
"The annotations listed" --> "The Annotations listed"
"documented annotations" --> "documented Annotations"

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +34,22 @@ Or
+
+The annotations listed under the `ingress.annotations` parameter are consumed as an array and any of the [documentend annotations for the k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#supported-ingress-annotations) may be used.
+
+When using the `virtual-server.f5.com/health` annotation the value must be a JSON array of the individual health monitors. Single and multiple health monitor examples can be seen in the [Ingress Examples](https://github.com/F5Networks/charts/tree/master/example_values) in this repo.
```
Annotation

recursivelycurious replied:
Done

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +34,22 @@ Or
+
+When using the `virtual-server.f5.com/health` annotation the value must be a JSON array of the individual health monitors. Single and multiple health monitor examples can be seen in the [Ingress Examples](https://github.com/F5Networks/charts/tree/master/example_values) in this repo.
+
+> CAUTION: Be sure to use the correct version of the Controller for the annotations you wish to use. See the [k8s-bigip-ctlr release notes](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/RELEASE-NOTES.html) for more information.
```
Annotations

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ctlr/README.md:
```diff
@@ -21,4 +33,23 @@ Or
+
+See the Controller documentation for a full list of [configuration parameters](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#controller-configuration-parameters).
+
 If you have a specific use case for F5 products in the kubernetes environment that would benefit from a curated chart, please [open an issue](https://github.com/F5Networks/charts/issues) describing your use case and providing example resources.
```
Kubernetes

recursivelycurious replied:
Done.

jputrino commented on src/incubator/f5-bigip-ingress/README.md:
```diff
@@ -20,4 +34,22 @@ Or
+
+> CAUTION: Be sure to use the correct version of the Controller for the annotations you wish to use. See the [k8s-bigip-ctlr release notes](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/RELEASE-NOTES.html) for more information.
+
 If you have a specific use case for F5 products in the kubernetes environment that would benefit from a curated chart, please [open an issue](https://github.com/F5Networks/charts/issues) describing your use case and providing example resources.
```
Kubernetes

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,55 @@
+
+See the Controller documentation for a full list of [configuration parameters](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#controller-configuration-parameters).
+
+If you have a specific use case for F5 products in the kubernetes environment that would benefit from a curated chart, please [open an issue](https://github.com/F5Networks/charts/issues) describing your use case and providing example resources.
```
Kubernetes

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,55 @@
+----------|-------------|--------
+bigip_login_secret | Secret that contains BIG-IP login credentials | f5-bigip-ctlr-login
+serviceaccount | name of ServiceAccount the ctlr should use | f5-bigip-ctlr-serviceaccount
+args.bigip_url | The admin IP for your BIG-IP | **Required**, no default
```
management IP

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,55 @@
+bigip_login_secret | Secret that contains BIG-IP login credentials | f5-bigip-ctlr-login
+serviceaccount | name of ServiceAccount the ctlr should use | f5-bigip-ctlr-serviceaccount
+args.bigip_url | The admin IP for your BIG-IP | **Required**, no default
+args.partition | BIG-IP partition the ctlr is to manage | f5-bigip-ctlr
```
"will manage" or "should manage"

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,55 @@
+args.bigip_url | The admin IP for your BIG-IP | **Required**, no default
+args.partition | BIG-IP partition the ctlr is to manage | f5-bigip-ctlr
+args.log_level | Log detail | DEBUG for incubation chart
+args.verify_interval | Interval in seconds to verify BIG-IP | 2 for incubation
```
Interval, in seconds, at which to verify BIG-IP settings

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ctlr/README.md:
```diff
@@ -0,0 +1,55 @@
+args.partition | BIG-IP partition the ctlr is to manage | f5-bigip-ctlr
+args.log_level | Log detail | DEBUG for incubation chart
+args.verify_interval | Interval in seconds to verify BIG-IP | 2 for incubation
+args.node_poll_interval | Interval in seconds to poll the cluster | 1 for incubation
```
Interval, in seconds, at which to poll the cluster

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ingress/README.md:
```diff
@@ -0,0 +1,55 @@
+----------|-------------|--------
+ingress.annotations.virtual-server.f5.com/ip | IP accepting traffic on the BIG-IP | **Required** no default
+ingress.annotations.virtual-server.f5.com/partition | BIG-IP partition of the Controller | **Required** no default
+ingress.namespace | Kubernetes/OpenShift namespace for the ingress | Optional
```
for the Ingress

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ingress/README.md:
```diff
@@ -0,0 +1,55 @@
+
+The annotations listed under the `ingress.annotations` parameter are consumed as an array and any of the [documentend annotations for the k8s-bigip-ctlr](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/#supported-ingress-annotations) may be used.
+
+When using the `virtual-server.f5.com/health` annotation the value must be a JSON array of the individual health monitors. Single and multiple health monitor examples can be seen in the [Ingress Examples](https://github.com/F5Networks/charts/tree/master/example_values) in this repo.
```
Annotation

recursivelycurious replied:
Done.

jputrino commented on src/stable/f5-bigip-ingress/README.md:
```diff
@@ -0,0 +1,55 @@
+
+> CAUTION: Be sure to use the correct version of the Controller for the annotations you wish to use. See the [k8s-bigip-ctlr release notes](http://clouddocs.f5.com/products/connectors/k8s-bigip-ctlr/latest/RELEASE-NOTES.html) for more information.
+
+If you have a specific use case for F5 products in the kubernetes environment that would benefit from a curated chart, please [open an issue](https://github.com/F5Networks/charts/issues) describing your use case and providing example resources.
```
Kubernetes

recursivelycurious replied:
Done.

recursivelycurious commented:
@jputrino - Addressed review comments, added as separate commit, and a third to "repackage, reindex" - both of which can be squashed in a "squash and merge".
